test/docker/kafka/consumer_simple_integration_test.go

```diff
@@ -107,3 +107,74 @@ func TestSimpleConsume_ClaimMessageError(t *testing.T) {
 		require.EqualError(t, err, "could not determine decoder  failed to determine content type from message headers [] : content type header is missing")
 	}
 }
+
+func TestSimpleConsume_WithDurationOffset(t *testing.T) {
+	now := time.Now()
+	sent := createTimestampPayload(
+		now.Add(-10*time.Hour),
+		now.Add(-5*time.Hour),
+		now.Add(-3*time.Hour),
+		now.Add(-2*time.Hour),
+		now.Add(-1*time.Hour),
+	)
+
+	messages := make([]*sarama.ProducerMessage, 0)
+	for _, val := range sent {
+		messages = append(messages, getProducerMessage(simpleTopic3, val))
+	}
+
+	err := sendMessages(messages...)
+	require.NoError(t, err)
+
+	chMessages := make(chan []string)
+	chErr := make(chan error)
+	go func() {
+		factory, err := simple.New("test1", simpleTopic3, Brokers(), kafka.DecoderJSON(), kafka.Version(sarama.V2_1_0_0.String()),
+			kafka.StartFromNewest(), kafka.WithDurationOffset(4*time.Hour, timestampExtractor))
+		if err != nil {
+			chErr <- err
+			return
+		}
+
+		consumer, err := factory.Create()
+		if err != nil {
+			chErr <- err
+			return
+		}
+		defer func() {
+			_ = consumer.Close()
+		}()
+
+		received, err := consumeMessages(consumer, 3)
+		if err != nil {
+			chErr <- err
+			return
+		}
+
+		chMessages <- received
+	}()
+
+	time.Sleep(5 * time.Second)
```

Comment:
it might be better if we avoided sleeps (some times a wait group or a mutex can help)
And we can also have a timeout on the test , as the failure condition ... 
WDYT ?

Reply:
I copied here the existing logic implemented in all the Kafka tests.
I agree that things could be improved by creating a consumer that would update a waitgroup and having a global timeout to limit the time execution. Yet, all the Kafka tests have do be refactored.
Are we okay to do it in this PR?

Reply:
It s fine, we can proceed with this as is.